### ISSUE TITLE
`show-gpus` show prices on different region/zones

### ISSUE THREAD

**Issue by Michaelvll** [skypilot-org/skypilot#1170]
It would be nice to show prices on different region/zones instead of per cloud based.

**Comment by vivekkhimani** [comment]
Hey @Michaelvll! I am currently investigating this issue and exploring various ways to go about implementing it. Currently, ```sky show-gpus``` supports ```--cloud``` option to show GPU availability filtered by the cloud provider. How do you feel about adding another option ```--region``` which will show the GPU availability by region? Also, we don't show pricing in the current output as well. Is adding a pricing column part of this feature request too?

**Comment by romilbhardwaj** [comment]
These are good questions @vivekkhimani! 

> How do you feel about adding another option --region which will show the GPU availability by region? 

I think it will be useful to clarify the how this flag is to be used. Each cloud has different names for the same geographical region (e.g., california is `us-west-1` on aws, `us-west2-<abcd>` on gcp, `westus` on azure). There are two modes:

1. `--region` can be specified only with `--cloud`. E.g. `sky show-gpus --cloud aws --region us-west-1`: This is easy. We can reject incorrect regions by checking if the region exists in the chosen cloud.
2. `--region` can be specified independent of `--cloud`. E.g. `sky show-gpus --region us-west-1`: This is a bit tricky. We could technically list only the regions in the AWS `us-west-1` region, but the user might be expecting all California resources across clouds to get listed. Supporting that use case is harder since we'll need to translate cloud-specific region codes to general geographic region codes. 

I am leaning towards 1, but I'll let @Michaelvll weigh in since he might have a better idea of the feature requirements.

> We don't show pricing in the current output as well. Is adding a pricing column part of this feature request too?

I think lets not include pricing for now. Users who want to see pricing can use `sky show-gpus -a`.

**Comment by Michaelvll** [comment]
Hey @vivekkhimani, it would be great if you want to take a look at this! I agree with @romilbhardwaj's comment that we may want to start from that the `--region <region-name>` needs to be with `--cloud`. 

Here are several requests:
1. `sky show-gpus V100`: display as the origin -- show the cheapest price for each cloud as following.
```
*NOTE*: for most GCP accelerators, INSTANCE_TYPE == (attachable) means the host VM's cost is not included.

GPU   QTY  CLOUD  INSTANCE_TYPE       vCPUs  HOST_MEMORY  HOURLY_PRICE  HOURLY_SPOT_PRICE  
V100  1    AWS    p3.2xlarge          8      61GB         $ 3.060       $ 0.918            
V100  4    AWS    p3.8xlarge          32     244GB        $ 12.240      $ 3.672            
V100  8    AWS    p3.16xlarge         64     488GB        $ 24.480      $ 7.344            
V100  1    Azure  Standard_NC6s_v3    6      112GB        $ 3.060       $ 0.846            
V100  2    Azure  Standard_NC12s_v3   12     224GB        $ 6.120       $ 1.693            
V100  4    Azure  Standard_NC24rs_v3  24     448GB        $ 13.460      $ 3.724            
V100  4    Azure  Standard_NC24s_v3   24     448GB        $ 12.240      $ 3.386            
V100  1    GCP    (attachable)        -      -            $ 2.480       $ 0.740            
V100  2    GCP    (attachable)        -      -            $ 4.960       $ 1.480            
V100  4    GCP    (attachable)        -      -            $ 9.920       $ 2.960            
V100  8    GCP    (attachable)        -      -            $ 19.840      $ 5.920            

GPU        QTY  CLOUD  INSTANCE_TYPE       vCPUs  HOST_MEMORY  HOURLY_PRICE  HOURLY_SPOT_PRICE  
V100-32GB  8    AWS    p3dn.24xlarge       96     768GB        $ 31.212      $ 9.364            
V100-32GB  8    Azure  Standard_ND40rs_v2  40     672GB        $ 22.032      $ 3.734 
```
2. `sky show-gpus V100 --cloud aws,gcp`: show the results in AWS and GCP and make the cloud name case insensitive. Note that currently there is bug for providing upper case cloud name and multiple clouds.
3. Add a `--all-zones [-z]` option (maybe can only be given when a accelerator is specified) to show all the prices in the per-zone way, for example:
```
> sky show-gpus V100 -z
GPU   QTY  CLOUD REGION(ZONE)  INSTANCE_TYPE       vCPUs  HOST_MEMORY  HOURLY_PRICE  HOURLY_SPOT_PRICE  
V100  1    AWS    us-east-1(us-east1-a) p3.2xlarge          8      61GB         $ 3.060       $ 0.918            
V100  1    AWS    us-east-1(us-east1-b) p3.2xlarge          8      61GB         $ 3.060       $ 0.918            
```

**Comment by vivekkhimani** [comment]
Thanks, @romilbhardwaj, and @Michaelvll for the clarification.

@romilbhardwaj I noticed the different naming conventions between different providers and was leaning towards your Option 2, which would include "normalizing" all of the cloud providers' own naming conventions for the regions. However, option 1 would be easier to implement and makes more sense for now. 

@Michaelvll thanks for the demo requests too. I was confused as I didn't see prices when I made a request without specifying an accelerator. However, this does make a lot of sense.

Thanks, I will put a PR together soon.
